File: Javascript/8kyu/count-of-positives-sum-of-negatives.js
//Decribtion:
// Given an array of integers.

// Return an array, where the first element is the count of positives numbers and the second element is sum of negative numbers. 0 is neither positive nor negative.

// If the input is an empty array or is null, return an empty array.
// Example

// For input [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, -11, -12, -13, -14, -15], you should return [10, -65].

//My Solution:
function countPositivesSumNegatives(input){
    if(input == null || input.length < 1){
      return [];
    }
    
    var newArray = [0, 0];
    
    for(var i = 0; i < input.length; i++){
      if(input[i] > 0)
        {
        newArray[0] += 1;
        }else {
        newArray[1] += input[i];
        }
      }
    return newArray;
    }

//Other solutions:
function countPositivesSumNegatives(input) {
    return input && input.length ? [input.filter(p => p > 0).length, input.filter(n => n < 0).reduce((a, b) => a + b, 0)] : [];
}

File: Python/8kyu/count-of-positives-sum-of-negatives.py
#Describtion:

#Solution:
def count_positives_sum_negatives(arr):
    n = (sum(num for num in arr if num < 0))
    
    x = 0
    
    for num in arr:
        if num > 0:
            x += 1
    
    if arr == []:
        return arr
    
    return [x, n]
    
#Other solutions:
def count_positives_sum_negatives(arr):
    if not arr: return []
    pos = 0
    neg = 0
    for x in arr:
      if x > 0:
          pos += 1
      if x < 0:
          neg += x
    return [pos, neg]